### DISCUSSION TITLE
ordered multi link

### DISCUSSION THREAD

**Discussion by hjalmarlucius** [geldata/gel#3271] [Q&A] [answered]
How should one best create an ordered multi link, e.g. for a time series of objects (in my case, an `array<float32>` plus some metadata). Would link properties be the correct way to go about it since ObjectType arrays are not permitted? Do you intend to permit ObjectType arrays at a later stage?

**Comment by elprans** [comment, marked as answer]
> Would link properties be the correct way to go about it

Yes, a link property is generally the way to go.

> Do you intend to permit ObjectType arrays at a later stage?

Unlikely.  It is quite hard to enforce referential integrity in collection types containing objects, which is the reason why we don't allow them at the schema level.  That said, object arrays are permitted in queries, so you can do `WITH obj_array := array_agg(Some.link ORDER BY @sort_order) ...` and work with the array in your query.
